lib/elixir/lib/calendar/datetime.ex

```diff
@@ -519,6 +519,12 @@ defmodule DateTime do
       {hour, minute, second} = unquote(read_time)
 
       cond do
+        offset == 0 ->
```

Comment:
Micro-optimization: what do you think about?

```elixir
cond do
  not calendar.valid_date?(year, month, day) ->
    {:error, :invalid_date}
 
  not calendar.valid_time?(hour, minute, second, microsecond) ->
    {:error, :invalid_time}

  offset == 0 ->
    datetime = %DateTime{
      calendar: calendar,
      year: year,
      month: month,
      day: day,
      hour: hour,
      minute: minute,
      second: second,
      microsecond: microsecond,
      std_offset: 0,
      utc_offset: 0,
      zone_abbr: "UTC",
      time_zone: "Etc/UTC"
    }

    {:ok, datetime, 0}

    # Rest...
```

Reply:
Sounds good.

Reply:
Wow. That made another ~18% improvement. Nice suggestion. Will push it up in a minute.